### PR TITLE
React/Dp-12303: InputSlider.

### PR DESCRIPTION
**Pull request by smurrayatwork**
## Description
Adds CompoundSlider, InputSlider, and Error components. Also adds the [react-compound-slider package](https://github.com/sghall/react-compound-slider). Adds FormContext support, changes the way Input renders to reduce re-renders on context changes.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-12303)

## Steps to Test
1. Pull the branch, run storybook, verify the atom at atoms/forms/InputSlider works.

## Screenshots
screencapture-localhost-6006-iframe-html-2019-01-29-12_27_50

### CODE REVIEW

File: react/src/components/atoms/forms/InputSlider/InputSlider.knobs.options.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export default {
+  labelText: () => text('InputSlider.labelText', 'Text Input'),
+  id: () => text('InputSlider.id', 'text-input'),
+  disabled: () => boolean('InputSlider.disabled', false),
+  defaultText: () => text('InputSlider.defaultText', '0'),
+  axis: () => select('InputSlider.axis', ['x', 'y'], 'x'),
+  max: () => number('InputSlider.max', 100),
+  min: () => number('InputSlider.min', 0),
+  step: () => number('InputSlider.step', 1),
+  ticks: () => [
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputSlider/InputSlider.knobs.options.js]: can you make this a knob that is editable? also can we default to label the min & max labels by the min and max values?

File: changelogs/DP-12303.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+___DESCRIPTION___
+Added
+Minor
+- (React) DP-12303: Adds CompoundSlider, InputSlider, Error components and adds the react-compound-slider package. Adds FormContext support, changes the way Input renders to reduce re-renders on context changes.
```
Comment (smurrayatwork) [inline changelogs/DP-12303.txt]:
```suggestion
- (React) DP-12303: Adds CompoundSlider, InputSlider, Error components and adds the react-compound-slider package. #431 
- (React) DP-12303: Adds FormContext support, changes the way Input renders to reduce re-renders on context changes. #431
```

File: react/src/components/atoms/forms/InputSlider/index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  /** Whether the label should be hidden or not */
+  hiddenLabel: PropTypes.bool,
+  /** The label text for the input field */
+  labelText: PropTypes.string.isRequired,
+  /** The unique ID for the input field */
+  id: PropTypes.string.isRequired,
+  /** Custom change function */
+  onChange: PropTypes.func,
+  /** Default input text value */
+  defaultText: PropTypes.string,
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputSlider/index.js]:
```suggestion
  defaultValue: PropTypes.string,
```

File: react/src/components/atoms/forms/InputSlider/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  axis: PropTypes.oneOf(['x', 'y']),
+  /** Disables the slider if true. */
+  disabled: PropTypes.bool,
+  /** Whether input is required or not */
+  required: PropTypes.bool
+};
+
+
+InputSlider.defaultProps = {
+  defaultText: 0,
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputSlider/index.js]:
```suggestion
  defaultValue: 0,
```

File: react/src/components/atoms/forms/InputSlider/InputSlider.knobs.options.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { text, boolean, number, object, select, array } from '@storybook/addon-knobs/react';
+
+export default {
+  labelText: () => text('InputSlider.labelText', 'Family Leave'),
+  id: () => text('InputSlider.id', 'text-input'),
+  disabled: () => boolean('InputSlider.disabled', false),
+  required: () => boolean('InputSlider.required', true),
+  defaultText: () => text('InputSlider.defaultText', '0'),
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/InputSlider/InputSlider.knobs.options.js]:
```suggestion
  defaultValue: () => text('InputSlider.defaultValue', '0'),
```

**Review by avrilpearl — APPROVED** [review]
LGTM

**Comment by clairesunstudio** [comment]
i added a focus style to the handle and added prop domain to override the default from inputSlider. one more thing is it needs to be able to handle float on keyboard up/down. otherwise looks good to me
screen shot 2019-01-30 at 11 56 48 am

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**